File: languages/dart/lib/ores_lib_core.dart
const redacted = '[REDACTED]';
const _sensitive = {'authorization','cookie','password','secret','token','access_token','refresh_token','id_token','client_secret','private_key','totp_seed','webauthn_challenge','face_image','face_template','fingerprint_image','fingerprint_template','biometric_template','voiceprint'};
String _compact(String value)=>value.trim().toLowerCase().replaceAll(RegExp(r'[^a-z0-9]'), '');
bool isSensitiveField(String key) { final normalized=_compact(key); return _sensitive.any((field)=>normalized==_compact(field)||normalized.endsWith(_compact(field))); }
Map<String,Object?> redactRecord(Map<String,Object?> value)=>value.map((key,item)=>MapEntry(key,isSensitiveField(key)?redacted:item));
bool validCorrelationId(String value)=>value.length>=8&&value.length<=128&&RegExp(r'^[A-Za-z0-9._:-]+$').hasMatch(value);
final class Secret<T> { const Secret(this._value); final T _value; R expose<R>(R Function(T) action)=>action(_value); @override String toString()=>redacted; }

File: languages/swift/Sources/OresLibCore/OresLibCore.swift
import Foundation
public let oresRedacted = "[REDACTED]"
private let sensitiveFields: Set<String> = ["authorization","cookie","password","secret","token","access_token","refresh_token","id_token","client_secret","private_key","totp_seed","webauthn_challenge","face_image","face_template","fingerprint_image","fingerprint_template","biometric_template","voiceprint"]
private func compact(_ value: String) -> String { value.lowercased().unicodeScalars.filter { CharacterSet.alphanumerics.contains($0) }.map(String.init).joined() }
public func isSensitiveField(_ key: String) -> Bool { let normalized=compact(key); return sensitiveFields.contains(where:{normalized == compact($0) || normalized.hasSuffix(compact($0))}) }
public func validCorrelationId(_ value: String) -> Bool { (8...128).contains(value.count) && value.unicodeScalars.allSatisfy { CharacterSet.alphanumerics.contains($0) || "._:-".unicodeScalars.contains($0) } }
public struct Secret<Value>: CustomStringConvertible { private let value: Value; public init(_ value:Value){self.value=value}; public func expose<Result>(_ action:(Value)->Result)->Result{action(value)}; public var description:String{oresRedacted} }
